### DISCUSSION TITLE
Delete old node Netdata Cloud

### DISCUSSION THREAD

**Discussion by nangavar** [netdata/netdata#12674] [Q&A] [answered]
Hi everyone.

Im not able to delete an old node that is offline now. Im within the General room.

When I try to delete it im prompted with "Room Cannot Be Changed - untouchable room"

Attached a screenshot of wat I got.

Any ideas? Thank you all a lot. I love netdata :)

<img width="407" alt="Screen Shot 2022-04-12 at 14 52 01" src="https://user-images.githubusercontent.com/66489890/163023962-feb38013-bcdb-4484-ba4c-74c5aa2ef9b4.png">

**Comment by ghost** [comment, marked as answer]
Hi @nangavar 

"General" room has been updated to include all nodes that exist in a space. It now works as a centralised node management directory, which was missing previously.

To preserve previous configuration. A copy of the general war room was created under "general-static".

Next, we already started working on the ability to soft delete nodes, so that they no longer appear across the account. This is different from the existing function to remove a node from a war room and will help with "removing" obsoleted lodes

**Reply by otavio** [reply]
So in summary the "General" cannot remove old nodes, YET. Right?

**Reply by ghost** [reply]
> So in summary the "General" cannot remove old nodes, YET. Right?

Yeap!

Working on a node deletion feature

**Comment by nangavar** [comment]
Thank you very much for your response! 👍

**Comment by mulder999** [comment]
Forget of obsolete space nodes would be very appreciated

**Comment by nderambure** [comment]
Any update on this ? thx !

**Comment by otavio** [comment]
Indeed; leaving old and unused nodes is very confusing.

**Comment by ghost** [comment]
This is work in progress. Shoud be available in the next couple of weeks

**Comment by TimFrayne** [comment]
Have been trying to find a way to remove unresponsive nodes and finally found this thread. Old tutorials from 2020 had me so confused. 

Glad to see it's being worked on.

**Comment by ottoort** [comment]
Any closer to being able to delete unreachable nodes?  
All this clutter is driving me crazy!

**Comment by ghost** [comment]
expecting to release it this week. fingers crossed

**Comment by johnrkriter** [comment]
I cannot believe that a product I have such tremendous respect for, and constantly tell my clients is the most holistic and manageable observability too, went back and REMOVED the ability to delete unused nodes and is STILL farting around with re-adding it.

If this does not get added soon, I might be back in the world of Nagios.

Get this fixed.

**Reply by andrewm4894** [reply]
@johnrkriter we have not removed the ability to delete unused nodes - you can still remove them from all spaces _apart_ from the default "General" space (which itself has confused a few users for sure). It's always been the case that you cannot delete nodes from the "General" space. What we did was default that all nodes go into "General" by default and can't be removed so "General" becomes like "Everything" so you always have one place where everything is by definition.

So what I do right now is make specific rooms for specific nodes and then you can just add/remove nodes to those rooms as needed. That way it's only the "General" room that can get very messy and cluttered with old nodes etc. So still annoying for sure, but at least its just that one place.

But I totally agree that we should have sorted this out a long time ago as it can cause such needless frustration for users - I hate it too when I can't clean up my general space :)

I believe we have the delete functionality on our testing and staging env's right now and should have it in next week or two.

p.s. thanks for spreading the word with clients!

**Comment by mulder999** [comment]
The best way I have found for cleaning, is to drop completely the whole space and enroll in a fresh new one. With load driven kubernetes/docker swarm it can quickly get messy, this is the fastest way to quickly restart clean once in a while.

**Reply by andrewm4894** [reply]
interesting. im defo keen to have an automated way to clean up and ideally some api's so users can do it programmatically too if better for them.

**Comment by lupimiguel** [comment]
![Remove_nodes1](https://user-images.githubusercontent.com/56279741/172447094-60d42a15-919d-4a6f-8cc2-f73c85919d89.gif)

This feature is live! We can all now enjoy the beauty of a clean UI to work on 😌 

Thanks to everyone for requesting this!

**Reply by johnrkriter** [reply]
Anyone have issues with actually clicking the button and having a successful delete? In my account when I click the button ,no action occurs. No console errors, and the event from the front end appears to go through. 

Functionality doesnt work.

**Reply by lupimiguel** [reply]
Thanks for reporting this so quickly. Another community member reported the same issue in Discord, and the FE team is looking at it. I'll let you know once they've found and fix it.

**Reply by johnrkriter** [reply]
@lupimiguel  I am happy to provide outside telemetry if required. if there are views or debug log I can provide happy to help

**Reply by lupimiguel** [reply]
@johnrkriter you're the best, thanks! I'm sure a FE dev or PM will come and ask if they need more info. We also have members reporting the button only showed up after clearing the cache, and that it's only working for nodes that are offline.

**Reply by mulder999** [reply]
This is so coooool ! It worked for me. Just had to Shift-F5 the page in Opera browser. Thanks @lupimiguel

**Reply by mulder999** [reply]
Just noticed the visited nodes still keep former references but that's relatively minor hindrance

**Reply by TimFrayne** [reply]
Thanks!! Was able to delete old unreachable nodes!

**Reply by burbuli8ra** [reply]
@johnrkriter we have made a fix on FE side, regarding this issue!
Could you please check that the functionality is indeed available for you as well?

**Reply by johnrkriter** [reply]
@burbuli8ra As of this moment, I still do not receive any UI pop-up when I click that button. 
I have tried clearing the entire cache in Firefox and Chrome, with no change. I have also tried disabling all adblock including my home DNS adblock(read:pihole)

My current setup

- OSX 12.4 Apple M1 Max
- Firefox 101.0 64-bit
- Chrome 102.0.5005.61 ARM64 official build
- Safari 15.5 (17613.2.7.1.8)

Setup 2
- Windows 10.0.19044
- Firefox 101.0 64-bit
- Chrome 102.0.5005.63 Official Build 64-bit

I have even tried using a cellular network but no change.

This is for space-id 05ab575a-c7c3-4519-89c0-1a8df943074f

**Reply by burbuli8ra** [reply]
Dear @johnrkriter thank you for providing all the extra info to us, which is indeed very helpful!
We have been able to reproduce the issue, so we will need some time to investigate it further and provide a proper solution (or even at least an estimation) for it ASAP!

**Reply by burbuli8ra** [reply]
@johnrkriter we have made a hotfix for this issue!
Could you please verify that you can indeed proceed with nodes obsoletion?

**Reply by johnrkriter** [reply]
@burbuli8ra I can confirm that even without clearing cache, all of the above browsers now properly display UI and allow item deletion. My issue is closed. 

Thank you for following up on this, and re-implementing a really critical feature to keep our sanity :)

**Reply by burbuli8ra** [reply]
Thank you too @johnrkriter for getting back to us and for providing the necessary info for solving this issue!
Glad we could help and that the functionality now works as expected! 😄

**Reply by msciacca** [reply]
I don't see that button at all in my general dashboard.

**Reply by cerealks** [reply]
same, no delete button here

**Reply by burbuli8ra** [reply]
@msciacca and @cerealks please note that the action for "deletion" is available only for admin users!
For members this action is not available and therefor the button is not visible as well!

**Reply by cerealks** [reply]
@burbuli8ra Thanks for the suggestion but I created the cloudaccount so I'm admin.

**Reply by msciacca** [reply]
@cerealks and @burbuli8ra I’ve noticed that the button to delete the node will only come up if it’s not communicating anymore with the Cloud, therefore if it’s offline.

**Reply by burbuli8ra** [reply]
@msciacca thank you for your reply!
@cerealks indeed only nodes with `Offline` status can be deleted!
If a node is `Online` or `Stale` this action is not available!
Also, please note that if you delete a node and it re-connects it will re-appear in the list of nodes! 😄

**Reply by cerealks** [reply]
Cheers @burbuli8ra thats right, I have a duplicate node which is in the stale state indeed. 
By stopping the service the node went to offline so I could remove the duplicate..

**Reply by blankster** [reply]
@burbuli8ra I face the problem, that I can't remove a stale node, as the server was reconnected with a new UUID. My result is that I only can remove the new one, but not the old one that seems to be stale "forever". Is there any option to get rid of it?

I've already tried to stop the node with the new UUID. Then I can remove the new one as it should be. But the old one can't be removed.

Let me know if you need more information and what would help you to help me.

**Comment by cakrit** [comment]
This functionality was greatly improved with yesterday's deployment. See https://github.com/netdata/netdata/issues/9358#issuecomment-1199437124 for details.

**Comment by inxbaby988** [comment]
I am sorry guys I try but seem like i cant delete my nodes we no longer need this but yet i try unable to delete it 
![Capture](https://user-images.githubusercontent.com/109713238/186369297-e3e744b0-5760-4801-bcf0-62a0c6e17628.JPG)
Apricated if anyone can help me out here thanks

**Reply by burbuli8ra** [reply]
Hello @inxbaby988 !
Currently the nodes deletion functionality is only available for nodes with "Offline" status and from what I see in the shared screenshot your node seems to be "Live"! Also, keep in mind that when a node is deleted when it reconnects to the cloud it automatically re-appears in the list of nodes!
Hope this info has been helpful for you!

**Comment by laurnts** [comment]
Hi Netdata, is there a possibility to remove old offline node via an API / CLI? We are running dynamic inventory that we automate. Hence we would like to have full automation in adding and removing node.

**Comment by hugovalente-pm** [comment]
Hi @laurnts ,

At the moment that action isn't possible since we don't have a publicly exposed API.
This is something we have on our backlog to provide, and our Nedata datasource Grafana plugin was a first step to exposing some endpoints publicly, but we don't have yet a date defined when to do this.

We have a sanitization script that removes offline nodes older than 60 days and there are talks to allow some sort of configuration per Space/War Room. Would something like this help until we have the public API available?
